schema_salad/typescript_codegen.py

```diff
@@ -279,9 +279,18 @@ def end_class(self, classname: str, field_names: List[str]) -> None:
             fieldtype = self.current_fieldtypes.get(safe_field_name)
             if fieldtype is None:
                 raise SchemaException(f"{safe_field_name} has no valid fieldtype")
+            if (
+                fieldtype.instance_type is not None
+                and "undefined" in fieldtype.instance_type
+            ):
+                optionalstring = "?"
+            else:
+                optionalstring = ""
```

Comment:
This commit produces the following, is that intended?

```
-  constructor ({extensionFields, loadingOptions, fields, type} : {extensionFields?: Dictionary<any>, loadingOptions?: LoadingOptions,  fields: undefined | Array<Internal.RecordField>, type: string,}) {
+  constructor ({extensionFields, loadingOptions, fields, type} : {extensionFields?: Dictionary<any>, loadingOptions?: LoadingOptions,  fields?: undefined | Array<Internal.RecordField>, type: string,}) {
```

Reply:
Pardon my typescript ignorance: seems so! https://www.typescriptlang.org/docs/handbook/2/objects.html

Reply:
Yeah! I would prefer if the `undefined` type would not be there for optional types (marked by `:?`) but I will have to put a bit more work in the type_loader function for that. It works just fine with the `undefined` though!